utils/helpers/format.ts

```diff
@@ -24,32 +25,39 @@ export const formatNumber = (num, precision, exponencial = false) =>
 
 //this function doesnt parse scientific notation floats, you need
 //to use toLocaleString if you want to avoid it
-export const floatToBN = (number, decimals = 18) => {
+export const floatToBN = (number?: number, decimals: number = 18): BigNumber | undefined => {
   try{
     if(!isEmpty(number)){
       return ethers.utils.parseUnits(roundDown(number,decimals),decimals);
     }else{
       return ethers.utils.parseUnits("0");
     }
   }catch(error){
-    console.error(error.message);
+    // TODO: remove later dvd: https://devblogs.microsoft.com/typescript/announcing-typescript-4-4/
+    if (error instanceof Error) {
+      console.error(error.message);
+    }
   }
 }
 
-export const BNToString = (bn, decimals = 18) => {
+export const BNToString = (bn: BigNumberish, decimals: string | BigNumberish = 18): string | undefined => {
   try{
-    return ethers.utils.formatUnits(bn,decimals).toLocaleString(undefined,{minimumFractionDigits:decimals});
+    return ethers.utils.formatUnits(bn, decimals).toLocaleString();
   }catch(error){
-    console.error(error.message);
+    if (error instanceof Error) {
+      console.error(error.message);
+    }
   }
 }
 
 //be aware that converting too big or too small numbers to float will
 //cause it to be converted to scientific notation
-export const BNToFloat = (bn, decimals = 18) => {
+export const BNToFloat = (bn: any, decimals: number = 18): number | undefined => {
   try{
     return Number(bn / 10 ** decimals);
```

Comment:
You are working with numbers here, would it be possible to give `bn` param a `type`?

Reply:
![image](https://user-images.githubusercontent.com/78897434/136302456-d5941189-4efd-407e-855a-bde365e8f5c1.png)

Reply:
![image](https://user-images.githubusercontent.com/78897434/136302456-d5941189-4efd-407e-855a-bde365e8f5c1.png)

Reply:
can you be more descriptive? just sharing screenshot do not answer the question I asked.

Reply:
If I set bn's type as BigNumberish, it shows above error like screenshot.
left-hand side of divide operator ( / ) must be type of 'any' or 'number'. That's why I set the type as 'any'